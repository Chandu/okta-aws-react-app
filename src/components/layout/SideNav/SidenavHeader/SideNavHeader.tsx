import { primaryTheme } from '@theme';
import { FC, memo } from 'react';
import * as Styled from './SideNavHeader.styled';

export const SideNavHeader: FC = memo(() => {
    return (
        <>
            <Styled.SideNavHeader>
                <Styled.DomainIcon size="S" variant="solid" iconColor={primaryTheme.colors.purple[500]} />
                <Styled.StyledText uppercase={true} textWeight="Medium">
                    OKTA-AWS-React App
                </Styled.StyledText>
            </Styled.SideNavHeader>
            <Styled.Separator />
        </>
    );
});

SideNavHeader.defaultProps = {};
SideNavHeader.displayName = 'SideNavHeader';
