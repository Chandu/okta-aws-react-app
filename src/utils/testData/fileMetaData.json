{
    "description": "dummy description",
    "uploaded-by-user-name": "xyz",
    "uploaded-timestamp": "Fri Apr 29 2022 18:46:10 GMT+0530 (India Standard Time)",
    "uploaded-user-email": "xyz@gmail.com"
}
