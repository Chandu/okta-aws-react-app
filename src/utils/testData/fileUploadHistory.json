[
    {
        "key": "1651756594298_dummyFile.csv",
        "description": "dummy description",
        "fileName": "1651756594298_dummyFile.csv",
        "uploaded-by-user-name": "xyz",
        "uploaded-timestamp": "Thu May 05 2022 19:28:07 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile1.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile1.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile3.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile3.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile4.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile4.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile5.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile5.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile6.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile6.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile7.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile7.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile8.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile8.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile9.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile9.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile10.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile10.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile11.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile11.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    }
]
