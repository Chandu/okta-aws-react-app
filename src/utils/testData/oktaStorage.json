{
    "idToken": {
        "claims": {
            "name": "dummy name",
            "email": "dummy email"
        },
        "idToken": "dummy token value"
    },
    "accessToken": {
        "accessToken": "dummy value"
    }
}
