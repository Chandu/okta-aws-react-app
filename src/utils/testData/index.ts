import fileMetaData from './fileMetaData.json';
import fileUploadHistory from './fileUploadHistory.json';
import oktaStorage from './oktaStorage.json';

export { oktaStorage, fileMetaData, fileUploadHistory };
