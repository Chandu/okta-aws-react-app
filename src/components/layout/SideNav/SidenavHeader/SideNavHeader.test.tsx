import { renderWithRouter } from '@test-utils';
import { oktaStorage } from '@utils/testData';
import { SideNavHeader } from './SideNavHeader';

const renderer = () => renderWithRouter(<SideNavHeader />);

describe('SideNavHeader', () => {
    beforeEach(() => localStorage.setItem('okta-token-storage', JSON.stringify(oktaStorage)));

    it('should render properly', () => {
        const { container } = renderer();
        expect(container).toMatchSnapshot();
    });
});
