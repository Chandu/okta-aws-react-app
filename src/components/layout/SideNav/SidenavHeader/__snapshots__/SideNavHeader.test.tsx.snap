// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavHeader should render properly 1`] = `
.c1 {
  overflow: visible;
  font-size: 2rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
  padding: 1rem;
  border-radius: 0.4rem;
  background-color: transparent;
}

.c1 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #6E14FF;
}

.c3 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 500;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2rem;
  text-align: initial;
  text-transform: uppercase;
  display: inline-block;
}

.c4 {
  margin-right: 1.4rem;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  text-transform: capitalize;
}

.c2 {
  z-index: inherit;
  border: 0.1px solid #C7D0D8;
  padding: 0.9rem;
  background-color: transparent;
}

.c2 > path {
  fill: #6E14FF;
}

.c0 {
  width: 100%;
  z-index: 3;
  min-height: 7.2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: center;
  display: grid;
  overflow: visible;
  grid-template-columns: 7.2rem calc(25.6rem - 7.2rem);
  color: #435465;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c5 {
  border-bottom: 0.1rem solid #dfe4e9;
  width: calc(100% - 4rem);
  margin: auto;
  margin-top: -0.1rem;
}

<div>
  <div
    class="c0"
  >
    <svg
      class="c1 c2"
      fill="none"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12 5v2h8c1.1 0 2 .9 2 2v10c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V5c0-1.1.9-2 2-2h6c1.1 0 2 .9 2 2zM4 19h2v-2H4v2zm2-4H4v-2h2v2zm-2-4h2V9H4v2zm2-4H4V5h2v2zm2 12h2v-2H8v2zm2-4H8v-2h2v2zm-2-4h2V9H8v2zm2-4H8V5h2v2zm2 12h7c.55 0 1-.45 1-1v-8c0-.55-.45-1-1-1h-7v2h2v2h-2v2h2v2h-2v2zm6-8h-2v2h2v-2zm-2 4h2v2h-2v-2z"
        fill="#000"
        fill-opacity=".54"
        fill-rule="evenodd"
      />
    </svg>
    <span
      class="c3 c4"
    >
      OKTA-AWS-React App
    </span>
  </div>
  <div
    class="c5"
  />
</div>
`;
